### PR TITLE
Arc Shape & Pie Charts

### PR DESCRIPTION
**Pull request by cedrickcooke**
Sorry about the variable names. I _truly_ don't know what's going on in the Arc math.

## Android
https://user-images.githubusercontent.com/20380774/151440541-bf68814c-949e-460f-97ab-959159bcb869.mp4

## Web
https://user-images.githubusercontent.com/20380774/151465060-462ddb57-2a3c-438c-9946-329647839949.mp4

### CODE REVIEW

**Review by davidtaylor-juul — APPROVED** [review]
Holy trigonometry Batman!

In theory you could give all those variables meaningful names. But if I drew it out I'd probably just end up with a bunch of theta, theta-prime, and gamma/phi gibberish for all the angles anyway. Naming the chord-lengths and corner treatments sensibly might be slightly useful, but personally I don't think it's worth it as it's unlikely to need any maintenance.

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -16,6 +16,7 @@ coroutines-js = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-core-js", v
 coroutines-test = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-test", version.ref = "coroutines" }
 datetime = { module = "org.jetbrains.kotlinx:kotlinx-datetime", version = "0.3.2" }
 robolectric = { module = "org.robolectric:robolectric", version = "4.7.3" }
+material = { module = "com.google.android.material:material", version = "1.5.0" }
```
Comment (twyatt) [inline gradle/libs.versions.toml:19]: nit: do you want to keep these alphabetically sorted?

Comment (cedrickcooke) [reply gradle/libs.versions.toml:19]: Good catch! Originally I meant for this dependency to be temporary, and forgot to fix this when I decided to leave it in for the sample.

File: shape/src/commonMain/kotlin/Arc.kt [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+    public fun outerRadius(value: Float): Arc = apply { outerRadius = value }
+    public fun innerRadius(value: Float): Arc = apply { innerRadius = value }
+    public fun cornerRadius(value: Float): Arc = apply { cornerRadius = value }
+    public fun padRadius(value: (Arc, Float, Float, Float) -> Float): Arc = apply { padRadius = value }
+
+    public operator fun invoke(slice: Slice<*>): Path = this(slice.startAngle, slice.endAngle, slice.padAngle)
+
+    public operator fun invoke(startAngle: Float, endAngle: Float, padAngle: Float): Path = Path {
+        // Implementation stolen from D3. https://github.com/d3/d3-shape/blob/main/src/arc.js#L87
```
Comment (twyatt) [inline shape/src/commonMain/kotlin/Arc.kt:47]: nit

Sounds friendlier as: 🙃 

```suggestion
        // Implementation "borrowed" from D3. https://github.com/d3/d3-shape/blob/main/src/arc.js#L87
```

**Review by twyatt — APPROVED** [review]

File: kanvas/src/jsMain/kotlin/Path2DBuilder.kt [hunk 3]
```diff
@@ -24,7 +24,8 @@ public class Path2DBuilder : RelativePathBuilder<Path2D>() {
         val y = top + radiusY
         val startAngleRadians = startAngle * PI / 180f
         val endAngleRadians = (startAngle + sweepAngle) * PI / 180f
-        buffer.ellipse(x, y, radiusX, radiusY, rotation = 0.0, startAngleRadians, endAngleRadians)
+        val anticlockwise = sweepAngle < 0f
+        buffer.ellipse(x, y, radiusX, radiusY, rotation = 0.0, startAngleRadians, endAngleRadians, anticlockwise)
```
[the remark this answers is not in the document]
Comment (cedrickcooke) [reply kanvas/src/jsMain/kotlin/Path2DBuilder.kt:28]: In general, I've been keeping documentation on the root interface, so the IDE will delegate quick-docs to the same central source of truth.

That said, apparently I missed explaining the units in the raw `PathBuilder` so I'll add it there.

**Pull request merged**